### PR TITLE
fix(reinvite-members): resending member invite does not modify member details

### PR DESCRIPTION
**Pull request by ameliahsu**
fix for https://github.com/getsentry/sentry/issues/80063
member details should not be modified by resending an invitation. to make sure no details are changed, we now return early once the invitation has been re-sent.

also added:
- an additional check that the request is not attempting to update member details and resend an invitation at the same time, since we don't want to allow this behavior
- a new audit log event for `MEMBER_REINVITE`, since reinviting doesn't really fall under `MEMBER_EDIT`

### CODE REVIEW

File: src/sentry/api/endpoints/organization_member/details.py [hunk 1]
```diff
@@ -48,7 +46,7 @@
 ERR_INSUFFICIENT_ROLE = "You cannot remove a member who has more access than you."
 ERR_INSUFFICIENT_SCOPE = "You are missing the member:admin scope."
 ERR_MEMBER_INVITE = "You cannot modify invitations sent by someone else."
-ERR_MEMBER_REINVITE = "You can only reinvite members; you cannot modify other member details."
+ERR_EDIT_WHEN_REINVITING = "You cannot modify other member details when resending an invitation."
```
Comment (sentaur-athena) [inline src/sentry/api/endpoints/organization_member/details.py:49]: This error doesn't show to our frontend ever and only API users will see it. What are they supposed to do if they want to modify the invite and resend? Should we add that context to the error message too?

File: src/sentry/api/endpoints/organization_member/details.py [hunk 2]
```diff
@@ -294,6 +275,25 @@ def put(
                 # TODO(dcramer): proper error message
                 return Response({"detail": ERR_UNINVITABLE}, status=400)
 
+            self.create_audit_entry(
```
Comment (sentaur-athena) [inline src/sentry/api/endpoints/organization_member/details.py:278]: 👍

**Review by sentaur-athena — APPROVED** [review]

File: tests/sentry/api/endpoints/test_organization_member_details.py [hunk 3]
```diff
@@ -264,6 +264,54 @@ def test_member_cannot_reinvite_members(self, mock_send_invite_email):
...
+            self.organization.slug, member_om.id, reinvite=1, role="manager", status_code=403
+        )
+        assert (
+            response.data.get("detail")
+            == "You cannot modify other member details when resending an invitation."
+        )
+        assert not mock_send_invite_email.mock_calls
+
+    @patch("sentry.models.OrganizationMember.send_invite_email")
+    def test_member_details_not_modified_after_reinviting(self, mock_send_invite_email):
```
Comment (iamrajjoshi) [inline tests/sentry/api/endpoints/test_organization_member_details.py:283]: extreme nit- maybe check for audit logs?

File: src/sentry/api/endpoints/organization_member/details.py (hunk 1 above)
Comment (ameliahsu) [reply src/sentry/api/endpoints/organization_member/details.py:49]: Would it be better to add something like "Separate requests must be made to modify member details and reinvite members"?

**Pull request merged**